enterprise/models.py

```diff
@@ -83,29 +94,160 @@ def __repr__(self):
         return self.__str__()
 
 
+class EnterpriseCustomerUserManager(models.Manager):
+    """
+    Model manager for :class:`.EnterpriseCustomerUser` entity.
+
+    This class should contain methods that create, modify or query :class:`.EnterpriseCustomerUser` entities.
+    """
+
+    @staticmethod
+    def _get_supported_filter_options(filter_options, model):
+        """
+        Filter out keyword arguments not supported by :method:`filter` method of `model`.
+
+        Arguments:
+            filter_options dict: filter options
+            model class: target model class
+        """
+        return {
+            key: value
+            for key, value in six.iteritems(filter_options)
+            if any(key.startswith(field_name) for field_name in get_all_field_names(model))
+        }
+
+    def list_linked_users(self, with_pending=False, **kwargs):
+        """
+        List linked users and (optionally) pending linked users.
+
+        Performs intellectual filtering of filter arguments, so only supported kwargs are passed to `filter` method.
+
+        Arguments:
+            with_pending (bool): include pending linked users?
+            kwargs (dict): keyword arguments passed to `filter` method
+        """
+        linked_users_kwargs = self._get_supported_filter_options(kwargs, EnterpriseCustomerUser)
+        linked_users = super(EnterpriseCustomerUserManager, self).filter(**linked_users_kwargs)
+
+        pending_linked_users_kwargs = self._get_supported_filter_options(kwargs, PendingEnterpriseCustomerUser)
+        if with_pending:
+            pending_linked_users = PendingEnterpriseCustomerUser.objects.filter(**pending_linked_users_kwargs)
+        else:
+            pending_linked_users = []
+
+        # list materializes the queryset, but we need it materialized for testing - chain creates a generator that
+        # is consumed at least once (when rendering a template), so by the time test gets to examine the context
+        # it contains exhausted generator. So, we're sacrificing laziness for repeatability
+        # TODO: find a way to retain queryset properties (laziness, repeatability)
```

Comment:
@mtyaka That's a good suggestion - I think it's worth implementing it now rather than delay it.